home-server-stub/pom.xml

```diff
@@ -11,7 +11,111 @@
 
+    <properties>
+        <yaml.name>openapi.yaml</yaml.name>
+    </properties>
+
     <artifactId>home-server-stub</artifactId>
 
+    <dependencies>
+        <dependency>
+            <groupId>com.softserveinc.ita.homeproject</groupId>
+            <artifactId>home-open-api</artifactId>
+            <version>${revision}</version>
+            <scope>provided</scope>
+        </dependency>
+    </dependencies>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-dependency-plugin</artifactId>
+                <version>2.8</version>
+                <executions>
+                    <execution>
+                        <id>resource-dependencies</id>
+                        <phase>generate-sources</phase>
+                        <goals>
+                            <goal>unpack</goal>
+                        </goals>
+                        <configuration>
+                            <artifactItems>
+                                <artifactItem>
+                                    <groupId>com.softserveinc.ita.homeproject</groupId>
+                                    <artifactId>home-open-api</artifactId>
+                                    <version>${revision}</version>
+                                    <type>jar</type>
+                                    <overWrite>true</overWrite>
+                                    <outputDirectory>${project.build.outputDirectory}</outputDirectory>
+                                </artifactItem>
+                            </artifactItems>
+                            <includes>*/${yaml.name}</includes>
+                            <outputDirectory>${project.build.outputDirectory}</outputDirectory>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+            <plugin>
+                <groupId>org.openapitools</groupId>
+                <artifactId>openapi-generator-maven-plugin</artifactId>
+                <version>4.3.1</version>
+                <executions>
+                    <execution>
+                        <phase>generate-sources</phase>
+                        <goals>
+                            <goal>generate</goal>
+                        </goals>
+                        <configuration>
+                            <inputSpec>${project.build.outputDirectory}/yaml/${yaml.name}</inputSpec>
+                            <generatorName>jaxrs-jersey</generatorName>
+                            <configOptions>
+                                <java8>true</java8>
+                                <dateLibrary>java8</dateLibrary>
+                                <legacyDiscriminatorBehavior>false</legacyDiscriminatorBehavior>
+                                <sourceFolder>src/gen/java/main</sourceFolder>
+                                <artifactId>home-server-stub</artifactId>
```

Comment:
?

Reply:
legacyDiscriminatorBehavior is redundant here.
dateLibrary is used here to use Java's datetime classes, if it wouldn't be here, codegen would use some random libarary.